codegen/package.go

```diff
@@ -256,36 +263,42 @@ func (p PackageHelper) DefaultMiddlewareSpecs() map[string]*MiddlewareSpec {
 	return p.defaultMiddlewareSpecs
 }
 
-// GenCodePackage returns the file path to the idl generated code folder
-func (p PackageHelper) GenCodePackage() string {
+// GenCodePackage returns map of idl type to the file path of the idl generated code folder
+func (p PackageHelper) GenCodePackage() map[string]string {
```

Comment:
probably also provide a lookup method that takes the string input, does lower case it and then look up the map. Can't safely assume the ext is always .proto and not .Proto or .PROTO

Reply:
I don't think `.Proto` and `.PROTO` are valid protobuf file extension. And as mentioned above the key is prescribed in the schema.